File: sdkwork-ai-sdk-flutter/lib/src/api/open.dart
import '../http/client.dart';
import '../models.dart';

class OpenApi {
  final HttpClient _client;
  
  OpenApi(this._client);

  Future<PlusApiResultListLabelOption?> getOptions(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/open/i18n/options'), params: params);
    return response is PlusApiResultListLabelOption ? response : null;
  }

  Future<PlusApiResultListLabelOption?> createOptions(Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/open/i18n/options'), params: params);
    return response is PlusApiResultListLabelOption ? response : null;
  }
}

File: sdkwork-ai-sdk-typescript/src/api/open.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultListLabelOption } from '../types';


export class OpenApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

async getOptions(params?: QueryParams): Promise<PlusApiResultListLabelOption> {
    return this.client.get<PlusApiResultListLabelOption>(aiApiPath(`/v1/open/i18n/options`), params);
  }

async createOptions(params?: QueryParams): Promise<PlusApiResultListLabelOption> {
    return this.client.post<PlusApiResultListLabelOption>(aiApiPath(`/v1/open/i18n/options`), undefined, params);
  }
}

export function createOpenApi(client: HttpClient): OpenApi {
  return new OpenApi(client);
}
